### PR TITLE
LPS-145775 Blueprints sidebar state does not persist when navigating between nav-links

### PR DESCRIPTION
**Pull request by thektan**
https://issues.liferay.com/browse/LPS-145775

Ended up using `sessionStorage` to avoid adding complexity by introducing another state or context that needed to be passed around.

### CODE REVIEW

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/utils/sessionStorage.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Helper function to set the session storage value of
+ * SESSION_IDS.ADD_SXP_ELEMENT_SIDEBAR.
+ * Toggles the state if `state` is undefined.
+ * @param {String} state Either 'open' or 'closed'.
+ */
+export function setItemAddSXPElementSidebar(state) {
+	if (!isDefined(state)) {
+		if (
+			sessionStorage.getItem(SESSION_IDS.ADD_SXP_ELEMENT_SIDEBAR) ===
```
Comment (oliv-yu) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/utils/sessionStorage.js:33]: Would it work to put a ternary operator here? If it makes it read easier. 🐸

Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/utils/sessionStorage.js:33]: Good idea, I can make this into

```
	sessionStorage.setItem(
		SESSION_IDS.ADD_SXP_ELEMENT_SIDEBAR,
		sessionStorage.getItem(SESSION_IDS.ADD_SXP_ELEMENT_SIDEBAR) ===
			'open'
			? 'closed'
			: 'open'
	);
```

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:search

**Comment by thektan** [comment]
ci:stop

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:search

**Comment by joshchong** [comment]
continued in https://github.com/liferay-search/liferay-portal/pull/333

**Pull request closed without merging**